include/exadg/fluid_structure_interaction/precice/driver_solid.h

```diff
@@ -169,18 +176,23 @@ class DriverSolid : public Driver<dim, Number>
       coupling_structure_to_fluid(structure->time_integrator->get_velocity_np(),
                                   structure->time_integrator->get_time_step_size());
 
-      // TODO: Add synchronization for the time-step size here. For now, we only allow a constant
-      // time-step size
       dealii::Timer precice_timer;
-      this->precice->advance(structure->time_integrator->get_time_step_size());
+      this->time_until_next_coupling_window =
+        this->precice->advance(structure->time_integrator->get_time_step_size());
       is_new_time_window = this->precice->is_time_window_complete();
       this->timer_tree.insert({"FSI", "preCICE"}, precice_timer.wall_time());
 
-      // Needs to be called before the swaps in post_solve
+      // Needs to be called before the swaps in post_solve.
       this->precice->reload_old_state_if_required([&]() {});
 
       if(is_new_time_window)
         structure->time_integrator->advance_one_timestep_post_solve();
+
+      // We need to adjust the time-step size in case we reach a coupling time-wimdow.
```

Comment:
window

Reply:
Actually no. The window refers to the step size of the coupled system, time-step size to the size of the solver.

Reply:
I mean the typo here. What do you mean?

Reply:
Ah ok sorry